internal/services/compute/snapshot_resource.go

```diff
@@ -241,6 +255,11 @@ func resourceSnapshotRead(d *pluginsdk.ResourceData, meta interface{}) error {
 			d.Set("create_option", string(data.CreateOption))
 			d.Set("storage_account_id", data.StorageAccountId)
 
+			if props.DiskAccessId != nil {
+				diskAccessID := *props.DiskAccessId
+				d.Set("disk_access_id", diskAccessID)
+			}
```

Comment:
We need to ensure a value is always set into the state, so can we make this:

```suggestion
			d.Set("disk_access_id", pointer.From(props.DiskAccessId))
```